rfc/005-generate-injection.md

```diff
@@ -50,7 +50,17 @@ To make `contrast generate` idempotent and handle potential edge cases, we will
 
 ### Service Mesh Injection
 
-The service mesh injection will follow a similar approach to the initializer injection outlined, possibly with additional configuration for labels for service discovery/ports. Details of the service mesh injection will be expanded in a future PR update to this RFC.
+The service mesh injection will follow a similar approach to the initializer injection outlined.
+During the generation process, all containers with the `contrast-cc` runtime class will have a service mesh added as a sidecar init container.
```

Comment:
I'm unsure whether service mesh should be enabled by default. Unlike the initializer, without which the deployment is somewhat pointless, I can live without a service mesh quite fine. How about making this opt-in until we have sufficient feedback from users that this should be default? We could also decide whether to inject based on presence of the annotations. cc @katexochen

Reply:
We would also need some form of configuration to inject a service mesh with default configuration, i.e. no ingress or egress. Maybe just include an annotation but leave it empty?

Reply:
A service mesh without ingress or egress means ... no service mesh? I'd consider "the default service mesh" to be no ingress exceptions and no egress setup, which is what you get when the sidecar is there but the env vars are empty.

Reply:
Yes, I meant no special ingress or egress configuration, so the default behavior of the service mesh. If we want to inject the service mesh based on the presence of annotations, we would need a way to tell the CLI to generate a service mesh, even if you don't specify ingress/egress in the annotations.

Reply:
There's a difference between the annotation being missing, or present but empty. I think taking the existence of one of the annotations as clue for injection would just work. 

There's another discussion about how to disable ingress completely while having egress, but that's unrelated to your proposal.